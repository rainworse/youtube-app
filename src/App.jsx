import LocalStorageDB from './backend/LocalStorageDB';
import Router from './frontend/Router';
import { useEffect, useState } from 'react';
import './frontend/styles/App.css';
import UserContext from './frontend/UserContext';
import { useBeforeUnload } from 'react-router-dom';

function App() {
  const [currentUser, setCurrentUser] = useState(null);

  useEffect(() => {
    // LocalStorageDB.initDB();
    setCurrentUser(LocalStorageDB.getUserSession());
  }, []);

  return (
    <>
      <UserContext.Provider
        value={{
          currentUser,
          setCurrentUser,
        }}
      >
        <Router />
      </UserContext.Provider>
    </>
  );
}

export default App;
