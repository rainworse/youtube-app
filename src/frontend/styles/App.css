@import url('https://fonts.googleapis.com/css2?family=Tinos:ital@0;1&display=swap');

* {
  margin: 0;
  padding: 0;
}

/* html,
body {
  height: 100%;
} */

body {
  font-family: 'Tinos', serif;
}
