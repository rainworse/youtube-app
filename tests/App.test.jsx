import { render, waitFor } from '@testing-library/react';
import { expect, vi } from 'vitest';
import App from '../src/App';

vi.mock('../src/backend/APIFetcher');

describe('App', () => {
  it('renders header and home page with two videos', async () => {
    const app = render(<App />);
    await waitFor(() => app.getByText('test video1'));
    expect(app.container).toMatchSnapshot();
  });
});
