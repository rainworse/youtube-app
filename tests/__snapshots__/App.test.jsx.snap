// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders App 1`] = `
<div>
  <div
    class="header"
  >
    <a
      class="header-logo-anchor"
      href="/"
    >
      <img
        alt="logo"
        class="header-logo"
        src="http://localhost:3000/public/assets/logo.svg"
      />
    </a>
    <form
      class="header-form"
    >
      <input
        class="header-search-input"
        type="text"
      />
      <input
        class="header-search-button"
        type="submit"
        value=""
      />
    </form>
    <div
      class="header-login-logout"
    >
      <div
        class="header-login-link"
      >
        <a
          class="header-login-anchor"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
  <div
    class="home-video-grid"
  >
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/001"
      >
        <img
          alt="test video1"
          class="video-card-thumbnail"
          src="non/existent/location1"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video1
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel001"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel1
            </div>
          </a>
        </div>
      </a>
    </div>
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/002"
      >
        <img
          alt="test video2"
          class="video-card-thumbnail"
          src="non/existent/location2"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video2
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel002"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel2
            </div>
          </a>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`App > renders header and home page with two videos 1`] = `
<div>
  <div
    class="header"
  >
    <a
      class="header-logo-anchor"
      href="/"
    >
      <img
        alt="logo"
        class="header-logo"
        src="http://localhost:3000/public/assets/logo.svg"
      />
    </a>
    <form
      class="header-form"
    >
      <input
        class="header-search-input"
        type="text"
      />
      <input
        class="header-search-button"
        type="submit"
        value=""
      />
    </form>
    <div
      class="header-login-logout"
    >
      <div
        class="header-login-link"
      >
        <a
          class="header-login-anchor"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
  <div
    class="home-video-grid"
  >
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/001"
      >
        <img
          alt="test video1"
          class="video-card-thumbnail"
          src="non/existent/location1"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video1
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel001"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel1
            </div>
          </a>
        </div>
      </a>
    </div>
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/002"
      >
        <img
          alt="test video2"
          class="video-card-thumbnail"
          src="non/existent/location2"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video2
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel002"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel2
            </div>
          </a>
        </div>
      </a>
    </div>
  </div>
</div>
`;
